crates/evm/src/executor/mod.rs

```diff
@@ -842,10 +843,16 @@ fn convert_call_result<D: Detokenize>(
 
     match status {
         return_ok!() => {
-            let result = decode_function_data(func, result, false)?;
+            let mut result = func.abi_decode_output(&result, false)?;
```

Comment:
I'm not sure using the dynamic encoder with the executor is the right way to approach this—as disambiguating with `DynSolValue` then becomes a tad cumbersome when using them. Do you think there's a way to use the static encoder here maybe with `SolValue` @DaniPopes ?

Reply:
Let's improve on this on a followup PR to not block on this—agree we should move to `alloy_sol_types`